### PR TITLE
Addressing Well Architected Framework gaps

### PR DESCRIPTION
**Pull request by Gordonby**
### Well Architected

Adding more coverage to the gaps we have highlighted by PSRule for the Well Architected Framework.
The only remaining one is the fact we need to add NSG's for the custom network, but that's out of scope for this PR because of some of the other networking feature branch work that is already underway in the DefaultDeny branch.

#### Changes

1. Splitting the PSRule WAF checks by GA/Preview
1. Bastion PublicIP to conditionally use AZ's
1. ACR zone redundancy (preview)
1. ACR untagged manifest retention (preview)
1. AKS diagnosticsettings, full category coverage
1. AKS Policy metadata
1. AKS [Disabling Local Accounts](https://azure.microsoft.com/en-us/updates/public-preview-create-aks-clusters-without-local-user-accounts-2/)

#### Accompanying UI changes to wizard
image

#### Results of the changes

**BEFORE** PSRule results from main branch.
> Rules processed: 64, failed: 10, errored: 0

**AFTER** PSRule results from PR
> GA Rules processed: 59, failed: 1, errored: 0
> GA+Preview Rules processed: 63, failed: 2, errored: 0

### Playwright regression test

Adding a regression test to cover the issue #147 which was fixed in #148
image

1. Creating a playwright test to export the AZ command to shell file
2. Adding additional steps to the GitHub Pages Publish action workflow to take the AZ CMD output and do a validation
3. On manual running of the GitHub workflow action, allow an optional full resource deployment attempt to our CI environment

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] This PR is ready to merge and is not **Work in Progress**
- [x] Link to a filed issue

### CODE REVIEW

File: helper/src/components/deployTab.js [hunk 1]
```diff
@@ -27,30 +27,31 @@ export default function DeployTab({ defaults, updateFn, tabValues, invalidArray,
...
+         custom_vnet: true,
+         ...serviceparams,
          ...aksvnetparams,
          ...(net.bastion !== defaults.net.bastion && {bastion: net.bastion}),
          ...(net.bastion && defaults.net.bastionSubnetAddressPrefix !== net.bastionSubnetAddressPrefix && {bastionSubnetAddressPrefix: net.bastionSubnetAddressPrefix})
        }),
     ...(net.vnet_opt === "byo" && { byoAKSSubnetId: net.byoAKSSubnetId, ...serviceparams }),
     ...(net.vnet_opt === "byo" && addons.ingress === 'appgw' && { byoAGWSubnetId: net.byoAGWSubnetId }),
     ...(cluster.enable_aad && { enable_aad: true, ...(cluster.enableAzureRBAC === false && cluster.aad_tenant_id && { aad_tenant_id: cluster.aad_tenant_id }) }),
+    ...(cluster.enable_aad && cluster.AksDisableLocalAccounts && { AksDisableLocalAccounts: cluster.AksDisableLocalAccounts }),
```
Comment (khowling) [inline helper/src/components/deployTab.js:40]: This is fine if the default is false, but will break if we switch the default to true,  maybe : ...(cluster.enable_aad && cluster.AksDisableLocalAccounts !== defaults.cluster.AksDisableLocalAccount && { AksDisableLocalAccounts: cluster.AksDisableLocalAccounts })

File: helper/src/components/deployTab.js [hunk 2]
```diff
@@ -262,7 +263,7 @@ ${cluster.apisecurity === "private" ? `az aks command invoke -g ${deploy.rg} -n
           <Separator ><div style={{ display: "flex", alignItems: 'center', }}><b style={{ marginRight: '10px' }}>Environment Name & Location</b></div> </Separator>
 
           <TextField label="Cluster Name" onChange={(ev, val) => updateFn('clusterName', val)} required errorMessage={getError(invalidArray, 'clusterName')} value={deploy.clusterName} />
-          <TextField label="Resource Group" onChange={(ev, val) => updateFn('rg', val)} required errorMessage={getError(invalidArray, 'rg')} value={deploy.rg} />
+          <TextField inputProps={{ "data-testid": "deploy-resourceGroup"}} id="azResourceGroup" label="Resource Group" onChange={(ev, val) => updateFn('rg', val)} required errorMessage={getError(invalidArray, 'rg')} value={deploy.rg} />
```
Comment (khowling) [inline helper/src/components/deployTab.js:266]: maybe dump the inputProp here on the TextField, its broken

File: helper/src/components/deployTab.js (hunk 1 above)
Comment (Gordonby) [reply helper/src/components/deployTab.js]: Makes sense. Thanks for the catch 👍

File: helper/src/components/deployTab.js (hunk 2 above)
Comment (Gordonby) [reply helper/src/components/deployTab.js:266]: i'll want to dump all of the instances then

Comment (Gordonby) [reply helper/src/components/deployTab.js]: Sorted.  Dumped all TextFields that attempt to use inputProps.

File: helper/src/components/deployTab.js (hunk 1 above)
Comment (Gordonby) [reply helper/src/components/deployTab.js]: Although @khowling - i totally just copied your extra statement and didn't notice the variable name was correct.
Thank goodness for Playwright.  It caught the behaviour of the checkbox -> az cmd wasn't right. 

🚀🚀🚀🚀🚀🚀

**Review by khowling — APPROVED** [review]

**Pull request merged**